src/data_collector/polygon_data/data_validator.py

```
@@ -51,24 +57,36 @@ def _serialize_timestamp(self, v, _info):
-
-        return ticker
+        """Validate ticker symbol format with security checks"""
+        try:
+            # Use secure validation from validation framework
+            return ValidationUtils.validate_ticker_symbol(v, "ticker")
+        except SecurityValidationError as e:
+            # Log security event and re-raise as ValueError for compatibility
+            log_input_validation_failure(
+                field="ticker", value=str(v)[:100], validation_error=str(e), severity="medium"
+            )
+            raise ValueError(f"Ticker validation failed: {e.message}")
```

Ruff (0.14.4)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)